api-definitions/common/s3-connection.yaml

```diff
@@ -0,0 +1,51 @@
+################################################################################
+# This schema defines the common components used to reference content held in a
+# cloud object store using an S3 interface.
+################################################################################
+
+schemas:
+  HMACCredentials:
+    type: object
+    properties:
+      access_key_id:
+        type: string
+        description: The public Access Key ID
+      secret_key:
+        type: string
+        description: The private Secret Key
+
+  IAMCredentials:
+    type: object
+    properties:
+      # TODO: What else goes here?
+      apikey:
+        type: string
+        description: The IAM apikey
```

Comment:
I believe both HMAC and IAM use the same concept of access_key and secret_key. If so, how about we consolidate both properties into one?

Something like?

```yaml
  Credentials:
    type: object
    description: Credentials for accessing the service
    properties:
      access_key:
        type: string
        description: The public access key
      private_key:
        type: string
        description: The private key
```

?

Reply:
I know at least for [IBM IAM](https://github.com/IBM/ibm-cos-sdk-python?tab=readme-ov-file#quick-start), there are different credentials (`apikey` + token endpoint + short-lived `token`). I think the AWS definition of IAM is different though, so maybe `IAM` is not a unique identifier.

The general question here would be whether we want to force `HMAC` and use that as the only supported access mechanism or try to keep this `oneOf` with an extensive list of provider-specific credential blob types.

Reply:
After reading the proposal a second time, I’m okay with it. I think I missed something during my initial review. 🤔